apps/src/blockly/addons/cdoUtils.ts

```diff
@@ -353,13 +378,16 @@ export function registerCustomProcedureBlocks() {
  * @param {string} customCategory The name of the custom category to retrieve blocks from. (Ex. 'VARIABLE', 'Behavior')
  * @returns {Document} A new XML document containing the filtered blocks.
  */
-export function getLevelToolboxBlocks(customCategory) {
+export function getLevelToolboxBlocks(customCategory: string) {
   const parser = new DOMParser();
+  if (!Blockly.toolboxBlocks) {
+    return;
+  }
   // TODO: Update this to support JSON once https://codedotorg.atlassian.net/browse/CT-8 is merged
   const xmlDoc = parser.parseFromString(Blockly.toolboxBlocks, 'text/xml');
 
   // Find the category based on the custom attribute
-  const categories = xmlDoc.getElementsByTagName('category');
+  const categories = Array.from(xmlDoc.getElementsByTagName('category'));
```

Comment:
We needed to use `Array.from` here (and a few other places) because TypeScript did not believe `categories` was an iterable type. The returned list is the same data as the original object.